otel-integration/k8s-helm/e2e-test/expected_test.go

```diff
@@ -0,0 +1,158 @@
+package e2e
+
+var expectedSchemaURL = map[string]bool{
+	"https://opentelemetry.io/schemas/1.6.1": false,
+	"https://opentelemetry.io/schemas/1.9.0": false,
+}
+
+const expectedScopeVersion = "0.93.0"
+
+var expectedScopeNames = map[string]bool{
+	"otelcol/hostmetricsreceiver/network":    false,
+	"otelcol/hostmetricsreceiver/cpu":        false,
+	"otelcol/hostmetricsreceiver/filesystem": false,
+	"otelcol/hostmetricsreceiver/memory":     false,
+	"otelcol/hostmetricsreceiver/load":       false,
+	"otelcol/hostmetricsreceiver/disk":       false,
+	"otelcol/kubeletstatsreceiver":           false,
+	"otelcol/prometheusreceiver":             false,
+}
+
+var expectedResourceAttributesKubeletstatreceiver = map[string]string{
+	"k8s.pod.uid":              "",
+	"k8s.pod.name":             "",
+	"k8s.namespace.name":       "",
+	"k8s.container.name":       "",
+	"k8s.daemonset.name":       "",
+	"k8s.deployment.name":      "",
+	"cx.otel_integration.name": "coralogix-integration-helm",
+	"k8s.cluster.name":         "otel-integration-agent-e2e",
+	"k8s.node.name":            "otel-integration-agent-e2e-control-plane",
+	"host.name":                "otel-integration-agent-e2e-control-plane",
+	"host.id":                  "",
+	"os.type":                  "linux",
+}
+
+var expectedResourceAttributesHostmetricsreceiver = map[string]string{
+	"cx.otel_integration.name": "coralogix-integration-helm",
+	"k8s.cluster.name":         "otel-integration-agent-e2e",
+	"k8s.node.name":            "otel-integration-agent-e2e-control-plane",
+	"host.name":                "otel-integration-agent-e2e-control-plane",
+	"host.id":                  "",
+	"os.type":                  "linux",
+}
+
+var expectedResourceAttributesPrometheusreceiver = map[string]string{
+	"service.name":             "opentelemetry-collector",
+	"net.host.name":            "",
+	"service.instance.id":      "",
+	"net.host.port":            "",
+	"http.scheme":              "http",
+	"service_version":          expectedScopeVersion,
+	"cx.otel_integration.name": "coralogix-integration-helm",
+	"k8s.cluster.name":         "otel-integration-agent-e2e",
+	"k8s.node.name":            "otel-integration-agent-e2e-control-plane",
+	"host.name":                "otel-integration-agent-e2e-control-plane",
+	"host.id":                  "",
+	"os.type":                  "linux",
+}
+
+var expectedMetrics map[string]bool = map[string]bool{
```

Comment:
That would be my preference as well. If we can figure out how to use https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/pdatatest to assert that would be great. Unfortunately I did not have luck with that since I could not make the expected / actual golden file results consistent between test runs.

Reply:
We could, though if it's just for our use, I find it easier to keep it in Go, as then we don't have to deal with (un)marshaling the data, i.e. it's one extra step to put this into a file. But if it's going to make it easier or enable people to maintain the tests without having to touch the code, I'm for it 👍.